### PR TITLE
Add case of passt

### PR DESCRIPTION
**Pull request by chloerh**
- VIRT-297034 - [user][passt] Passt backend interface - check vm interface info and network connectivity
- VIRT-297101 - [user][passt] Passt backend interface - check the portforwarding feature

Depends on:
- https://github.com/avocado-framework/avocado-vt/pull/3667
- https://github.com/avocado-framework/avocado-vt/pull/3668
- https://github.com/avocado-framework/avocado-vt/pull/3673
- https://github.com/avocado-framework/avocado-vt/pull/3676
- https://github.com/avocado-framework/avocado-vt/pull/3678

Test result:
 (1/4) type_specific.io-github-autotest-libvirt.virtual_network.passt.interface_function.minimal.root_user: PASS (249.55 s)
 (2/4) type_specific.io-github-autotest-libvirt.virtual_network.passt.interface_function.minimal.non_root_user: PASS (175.07 s)
 (3/4) type_specific.io-github-autotest-libvirt.virtual_network.passt.interface_function.ip_portfw.root_user: PASS (555.58 s)
 (4/4) type_specific.io-github-autotest-libvirt.virtual_network.passt.interface_function.ip_portfw.non_root_user: PASS (419.26 s)
RESULTS    : PASS 4 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /var/lib/avocado/job-results/job-2023-04-26T23.27-943971c/results.html
JOB TIME   : 1400.89 s

### CODE REVIEW

File: libvirt/tests/src/virtual_network/passt/passt_function.py [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+
+        ips = {
+            'outside_ip': outside_ip,
+            'host_public_ip': host_ip,
+        }
+        network_base.ping_check(params, ips, session, force_ipv4=True)
+        session.close()
+
+        firewalld.stop()
+        LOG.debug(firewalld.status())
```
Comment (Yingshun) [inline libvirt/tests/src/virtual_network/passt/passt_function.py:118]: Can give a detailed log information?

File: provider/virtual_network/passt.py [hunk 2]
```diff
@@ -0,0 +1,343 @@
...
+    Get a random free port
+
+    :return: port id
+    """
+    with socket() as s:
+        s.bind(('', 0))
+        return s.getsockname()[1]
+
+
+def get_iface_info(iface='', session=None):
```
Comment (Yingshun) [inline provider/virtual_network/passt.py:38]: I think this function can be used by other tests, so I wonder if you can move it to utils_net.py or enhace utils_net.get_linux_iface_info()? what do you think?

File: provider/virtual_network/passt.py [hunk 3]
```diff
@@ -0,0 +1,343 @@
...
+    msg = f'Message {utils_misc.generate_random_string(6)} '\
+          f'from source via {protocol}'
+    if src_iface:
+        addr = f'{addr}%{src_iface}'
+    if ':' in addr:
+        addr = f'[{addr}]'
+    cmd = f'echo "{msg}" | socat -u STDIN {protocol}:{addr}:{src_port}'
+
+    src_output = src_sess.cmd_output(cmd)
+    LOG.debug(src_output)
```
Comment (Yingshun) [inline provider/virtual_network/passt.py:259]: Can give a detailed log information?

**Review by yalzhang — APPROVED** [review]
The checkpoints are good.

File: libvirt/tests/src/virtual_network/passt/passt_function.py (hunk 1 above)
Comment (chloerh) [reply libvirt/tests/src/virtual_network/passt/passt_function.py]: Done

File: provider/virtual_network/passt.py (hunk 2 above)
Comment (chloerh) [reply provider/virtual_network/passt.py]: Sure, Updated.

File: provider/virtual_network/passt.py (hunk 3 above)
Comment (chloerh) [reply provider/virtual_network/passt.py]: Updated

**Review by Yingshun — APPROVED** [review]
LGTM

**Review by Yingshun — APPROVED** [review]
LGTM

**Pull request merged**